solana/ts/src/swapLayer/relayerFees.ts

```diff
@@ -0,0 +1,125 @@
+import { RelayParams } from ".";
+
+const EVM_GAS_OVERHEAD = 100_000n;
+const DROPOFF_GAS_OVERHEAD = 10_000n;
+const UNISWAP_GAS_OVERHEAD = 100_000n;
+const UNISWAP_GAS_PER_SWAP = 100_000n;
+const TRADERJOE_GAS_OVERHEAD = 100_000n;
+const TRADERJOE_GAS_PER_SWAP = 100_000n;
+const ONE_SOL = 1_000_000_000n;
+const GAS_PRICE_SCALAR = 1_000_000n;
+const GAS_DROPOFF_SCALAR = 1_000n;
+const ONE_ETHER = 1_000_000_000_000_000_000n;
+const U32_MAX = 4294967295;
+const MAX_BPS = 1_000_000n;
+
+// TODO: Need to implement serde for the following types. For now, we will use a dummy
+// type to represent the swap type.
+export type SwapType = {
+    none?: {};
+    uniswapV3?: {};
+    traderJoe?: {};
+    jupiterV6?: {};
+};
+
+export function denormalizeGasDropOff(gasDropOff: number): bigint {
+    return BigInt(gasDropOff) * GAS_DROPOFF_SCALAR;
```

Comment:
is gasDropOff always a whole number?

Reply:
Yes, that's correct. It's specified as micro-ether by the user. Regardless of the sending chain's native decimals, 1e6 will always represent one native token (e.g., one ether). However, in the contract we have to "denormalize" the specified amount to reflect the actual value to compute the correct relayer fee in USDC terms. That's why this function multiplies the `gasDropOff` amount by 1e3.